develop/components/common/TitleBar.tsx

```diff
@@ -109,18 +110,33 @@ export default class TitleBar extends React.Component<TitleBarProps, TitleBarSta
         this.setState({ searchInFocus: false, searchList: [] });
     }
 
+    /**
+     * Checks whether the input key is an `Escape` key
+     * @param {React.KeyboardEvent<HTMLInputElement>} event The keyboard event
+     * @returns {boolean} True if it is `Escape` key
+     */
+    isEscapeKey(event: React.KeyboardEvent<HTMLInputElement>): boolean {
+        if (event.key === "Escape" || event.key === "Esc") {
+            return true;
+        } else if (event.keyCode === 27) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
     componentDidMount() {
         this.setState({ componentsList: [].concat(...[sidebarData.form, sidebarData.ui, sidebarData.other]) }, () => {
             console.log("Number of Components: ", this.state.componentsList.length);
         });
 
-        document.addEventListener("keypress", (e: KeyboardEvent) => {
+        const onKeyPress: EventListener = (e: KeyboardEvent) => {
```

Comment:
I had take this outside `componentDidMount `and have it within the class like 

     ```
   onKeyUp = (e: KeyboardEvent) => {
            if (document.activeElement === this.searchRef.current) {
                if (e.key === "Escape" || e.keyCode === 27) {
                    if (document.activeElement === this.searchRef.current) {
                        this.setState({ searchTerm: "", searchList: [], searchInFocus: false }, () => (this.searchRef.current as HTMLInputElement).blur());
                    }
                }
                if (e.key === "ArrowUp" || e.keyCode === 38) {
                    e.preventDefault();
                    if (this.state.highlighted > 0) {
                        this.setState({ highlighted: this.state.highlighted - 1 });
                    }
                }
                if (e.key === "ArrowDown" || e.keyCode === 40) {
                    e.preventDefault();
                    if (this.state.highlighted < (this.state.searchList.length - 1)) {
                        this.setState({ highlighted: this.state.highlighted + 1 });
                    }
                }
            }
        }

      onKeyPress: EventListener = (e: KeyboardEvent) => {
            if (e.ctrlKey && e.shiftKey && (e.key.toLowerCase() === "f" || e.keyCode === 6)) {
                this.searchRef.current.focus();
            }
        }
      componentDidMount() {
        this.setState({ componentsList: [].concat(...[sidebarData.form, sidebarData.ui, sidebarData.other]) }, () => {
            console.log("Number of Components: ", this.state.componentsList.length);
        });

        document.addEventListener("keyup", this.onKeyUp);
        document.addEventListener("keypress", this.onKeyPress);
    }
```

And then 

componentWillMount() {
        document.removeEventListener("keyup", this.onKeyUp);
        document.removeEventListener("keypress", this.onKeyPress);
}

Reply:
Although there is no need to remove the listener since the only scenario where this component will unmount is when the user leaves the site, but I agree it's a good practice to have, so I added it. Thanks for reminding me.